### PR TITLE
Switch to Rails 6.1

### PR DESCRIPTION
**Pull request by iHiD**

### CODE REVIEW

File: test/commands/mentor/reply_to_discussion_test.rb [hunk 1]
```diff
@@ -31,7 +31,7 @@ class Mentor::ReplyToDiscussionTest < ActiveSupport::TestCase
       "foobar"
     )
     assert_equal 1, user.notifications.size
-    notification = user.notifications.first
+    notification = Notification.where(user: user).first
```
Comment (iHiD) [inline test/commands/mentor/reply_to_discussion_test.rb:34]: These are from me working on getting strict_loading working. It's problematic with FactoryBot so I've stopped for now. But these are still fine to merge.

File: app/commands/iteration/create.rb [hunk 2]
```diff
@@ -5,9 +5,10 @@ class Create
     initialize_with :solution, :submission
 
     def call
+      time = Time.now.utc.to_s(:db)
```
Comment (kntsoriano) [inline app/commands/iteration/create.rb:8]: Nice.

File: test/commands/mentor/reply_to_discussion_test.rb (hunk 1 above)
Comment (kntsoriano) [reply test/commands/mentor/reply_to_discussion_test.rb:34]: What's strict loading?

**Review by kntsoriano — APPROVED** [review]

File: test/commands/mentor/reply_to_discussion_test.rb (hunk 1 above)
Comment (iHiD) [reply test/commands/mentor/reply_to_discussion_test.rb:34]: It basically causes an exception if an `n+1` might happen.

https://github.com/rails/rails/pull/37400
https://github.com/rails/rails/pull/39491

File: app/commands/iteration/create.rb (hunk 2 above)
Comment (iHiD) [reply app/commands/iteration/create.rb:8]: `NOW()` doesn't work with the latest version of MySQL on mac for some reason.

**Pull request merged**